src/Microsoft.DotNet.Darc/src/DarcLib/MaestroApiBarClient.cs

```diff
@@ -302,28 +302,9 @@ public async Task SetRepositoryMergePoliciesAsync(string repoUri, string branch,
                                                  int? buildId = null,
                                                  bool? nonShipping = null)
         {
-            // Start at the first page and go until we 404
-            List<Asset> assets = new List<Asset>();
-            int page = 0;
-            while (true)
-            {
-                try
-                {
-                    IReadOnlyList<Asset> assetPage = await _barClient.Assets.ListAssetsAsync(
-                        name: name,
-                        version: version,
-                        buildId: buildId,
-                        nonShipping: nonShipping,
-                        loadLocations: true,
-                        page: ++page);
-                    assets.AddRange(assetPage);
-                }
-                catch (RestApiException e) when (e.Response.StatusCode == HttpStatusCode.NotFound)
-                {
-                    break;
-                }
-            }
-            return assets;
+            PagedResponse<Asset> pagedResponse = await _barClient.Assets.ListAssetsAsync(name: name,
```

Comment:
NIT: not sure if these parameter names "name: name, version: version" are adding much value here :-)